components/content-service/cmd/run.go

```diff
@@ -32,106 +19,48 @@ var runCmd = &cobra.Command{
 
 	Run: func(cmd *cobra.Command, args []string) {
 		cfg := getConfig()
-		reg := prometheus.NewRegistry()
-
-		common_grpc.SetupLogging()
 
-		grpcMetrics := grpc_prometheus.NewServerMetrics()
-		grpcMetrics.EnableHandlingTimeHistogram()
-		reg.MustRegister(grpcMetrics)
-
-		grpcOpts := common_grpc.ServerOptionsWithInterceptors(
-			[]grpc.StreamServerInterceptor{grpcMetrics.StreamServerInterceptor()},
-			[]grpc.UnaryServerInterceptor{grpcMetrics.UnaryServerInterceptor()},
+		srv, err := baseserver.New("content-service",
+			baseserver.WithGRPC(&cfg.Service),
 		)
-		tlsOpt, err := cfg.Service.TLS.ServerOption()
 		if err != nil {
-			log.WithError(err).Fatal("cannot use TLS config")
-		}
-		if tlsOpt != nil {
-			log.WithField("crt", cfg.Service.TLS.Certificate).WithField("key", cfg.Service.TLS.PrivateKey).Debug("securing gRPC server with TLS")
-			grpcOpts = append(grpcOpts, tlsOpt)
-		} else {
-			log.Warn("no TLS configured - gRPC server will be unsecured")
+			log.WithError(err).Fatal("Failed to create server.")
 		}
 
-		server := grpc.NewServer(grpcOpts...)
-
 		contentService, err := service.NewContentService(cfg.Storage)
 		if err != nil {
-			log.WithError(err).Fatalf("cannot create content service")
+			log.WithError(err).Fatalf("Cannot create content service")
```

Comment:
please leave this lowercase. The vast majority of our error messages is lowercase, and we should keep things consistent

Reply:
General practice in Go (but elsewhere) is to uppercase logs to make them sentences. They are designed to be consumed by humans and first letter uppercase is slightly easier to visually grep. Even our style guide uses first letter uppercase in logs https://github.com/uber-go/guide/blob/master/style.md#reduce-nesting

The change here is really just a drive by improvement to be more consistent with general practice.

Errors, on the other hand, should never start with an uppercase because they get wrapped and composed so don't really form sentences.

Reply:
Here's another reference from [go code review style guide](https://github.com/golang/go/wiki/CodeReviewComments#error-strings)

Reply:
Fair point.

/hold cancel